ingestion-lambda/src/handler.ts

```diff
@@ -31,45 +63,64 @@ function processMessage(
 	};
 }
 
-function isSESEvent(event: SQSEvent | SESEvent): event is SESEvent {
-	return event.Records.some((record) => 'ses' in record);
+function processRecord(
+	record: SESEventRecord | SQSRecord,
+): OperationResult<{ externalId: string; objectKey: string }> {
+	if (isSESRecord(record)) {
+		return processSESRecord(record);
+	} else {
+		return processSQSRecord(record);
+	}
+}
+
+function isSESRecord(
+	record: SESEventRecord | SQSRecord,
+): record is SESEventRecord {
+	return 'ses' in record;
```

Comment:
Is this fully robust? Could an sqs record that has the string ses pass this check? Might be unlikely but could be worth seeing if there is a specific field to key from.

Reply:
Ok cool 👍